examples/Linux/sentry.c

```diff
@@ -0,0 +1,324 @@
+    ret = wolfsentry_route_event_dispatch_with_inited_result(
+            wolfsentry,
+            &remote.sa,
+            &local.sa,
+            WOLFSENTRY_ROUTE_FLAG_DIRECTION_IN,
+            "call-in-from-echo",
+            strlen("call-in-from-echo"),
+            NULL,
+            NULL,
+            NULL,
+            &action_results);
+
+    printf("TCP Sentry return: %d\n", ret);
+    fflush(stdout);
+
+    /* Check the result, if it contains "reject" then notify the caller */
+    if (WOLFSENTRY_ERROR_DECODE_ERROR_CODE(ret) >= 0) {
+        if (WOLFSENTRY_MASKIN_BITS(action_results, WOLFSENTRY_ACTION_RES_REJECT)) {
+            return -1;
+        }
+    }
+
+    return 0;
+}
+
+/* Check / validate ICMP traffic */
+int sentry_action_ping(const ip_addr_t *addr, u8_t type)
+{
+    wolfsentry_errcode_t ret;
+    wolfsentry_action_res_t action_results;
+    /* As above, pad the struct to make addr 4 bytes */
+    struct {
+        struct wolfsentry_sockaddr sa;
+        byte addr_buf[4];
+    } remote;
+
+    /* ICMP protocol check. The ICMP packet type is the port number for this
+     * check */
+    remote.sa.sa_family = WOLFSENTRY_AF_INET;
+    remote.sa.sa_proto = IPPROTO_ICMP;
+    remote.sa.sa_port = type;
+    remote.sa.addr_len = 32;
+    remote.sa.interface = 0;
+    memcpy(remote.sa.addr, &addr->addr, 4);
+
+    ret = wolfsentry_route_event_dispatch(
+            wolfsentry,
+            &remote.sa,
+            &remote.sa, // Reuse for now
+            WOLFSENTRY_ROUTE_FLAG_DIRECTION_IN,
+            "call-in-from-echo",
+            strlen("call-in-from-echo"),
+            NULL,
+            NULL,
+            NULL,
+            &action_results);
+
+    printf("PING Sentry action return: %d\n", ret);
+    fflush(stdout);
+    if (WOLFSENTRY_ERROR_DECODE_ERROR_CODE(ret) >= 0) {
+        if (WOLFSENTRY_MASKIN_BITS(action_results, WOLFSENTRY_ACTION_RES_REJECT)) {
+            return -1;
+        }
+    }
+
+    return 0;
+}
+
+/* Check MAC address */
+int sentry_action_mac(struct eth_addr *addr)
+{
+    wolfsentry_errcode_t ret;
+    wolfsentry_action_res_t action_results;
+    /* Pad addr to 6 bytes for the hardware address */
+    struct {
+        struct wolfsentry_sockaddr sa;
+        byte addr_buf[6];
+    } remote;
+
+    /* We only really care about the data and length, the family is AF_LINK */
+    remote.sa.sa_family = WOLFSENTRY_AF_LINK;
+    remote.sa.addr_len = 48;
+    remote.sa.interface = 0;
+    // MAC addresses are 6 bytes (48 bits)
+    memcpy(remote.sa.addr, &addr->addr, 6);
+
+    ret = wolfsentry_route_event_dispatch(
+            wolfsentry,
+            &remote.sa,
+            &remote.sa, // Reuse for now
+            WOLFSENTRY_ROUTE_FLAG_DIRECTION_IN,
+            "call-in-from-echo",
+            strlen("call-in-from-echo"),
+            NULL,
+            NULL,
+            NULL,
+            &action_results);
+
+    printf("MAC Sentry action return: %d\n", ret);
```

Comment:
should use `WOLFSENTRY_ERROR_FMT` and `WOLFSENTRY_ERROR_FMT_ARGS()` to make these return codes legible.